reference/conanfile/methods/compatibility.rst

```diff
@@ -0,0 +1,46 @@
+.. _reference_conanfile_methods_compatibility:
+
+compatibility()
+===============
+
+.. warning::
+
+    This is a **preview** feature
+
+
+The ``compatibility()`` method implements the same binary compatibility mechanism than the
+:ref:`compatibility plugin<reference_extensions_binary_compatibility>`, but at the recipe
+level. In general, the global compatibility plugin should be good for most cases, and only
+require the recipe method for exceptional cases.
+
+This method can be used in a *conanfile.py* to define packages that are compatible between
+each other. If there are not binaries available for the requested settings and options
+this mechanism will retrieve the compatible packages' binaries if they exist.  The method
+should return a list of compatible configurations.
+
+For example, if we want that binaries
+built with gcc versions 4.8, 4.7 and 4.6 are considered compatible with the ones compiled
+with 4.9 we could declare the ``compatibility()`` like this:
+
+..  code-block:: python
+
+    def compatibility(self):
+        if self.settings.compiler == "gcc" and self.settings.compiler.version == "4.9":
+            return [{"settings": [("compiler.version", v)]}
```

Comment:
this doesn't look like a simple list, it's a list of dictionaries of lists of dictionaries :thinking: 
I think it deserves a little bit more explanation than just `list of compatible configurations`.
a static returned structure is also more illustrative for an example, than some code that uses list comprehension (it's kinda hard to understand that's the hierarchy, structure and type of the returned entity).

Reply:
It is good enough for this in this location, it will get a full explanation in the plugin.